homeassistant/components/tradfri/sensor.py

```diff
@@ -76,18 +143,26 @@ async def async_setup_entry(
     entities: list[TradfriSensor] = []
 
     for device_coordinator in coordinator_data[COORDINATOR_LIST]:
-        description = None
         if (
             not device_coordinator.device.has_light_control
             and not device_coordinator.device.has_socket_control
             and not device_coordinator.device.has_signal_repeater_control
             and not device_coordinator.device.has_air_purifier_control
         ):
-            description = SENSOR_DESCRIPTION_BATTERY
+            descriptions = SENSOR_DESCRIPTIONS_BATTERY
         elif device_coordinator.device.has_air_purifier_control:
-            description = SENSOR_DESCRIPTION_AQI
+            descriptions = SENSOR_DESCRIPTIONS_FAN
+        else:
+            continue
+
+        for description in descriptions:
+            # Added in Home assistant 2022.3
+            _migrate_old_unique_ids(
```

Comment:
Can we add a test for the unique_id migration? Here's an example:
https://github.com/home-assistant/core/blob/63d3a47599becc313c11b0bdebbded8a5a3f35f0/tests/components/zwave_js/test_migrate.py#L495-L534

The action for Tradfri that should do the migration is just setting up the integration.

Reply:
Of course, I’ll look into this tomorrow.

Reply:
I this how you meant?